src/utils/http.ts

```diff
@@ -1,11 +1,17 @@
 import { Decoder } from 'io-ts/Decoder';
 import { pipe } from 'fp-ts/function';
-import { Either, isRight } from 'fp-ts/lib/Either';
+import { Either, isRight, isLeft } from 'fp-ts/lib/Either';
 
 const parse =
-  <A, B>(model: Decoder<A, B>) =>
-  (data: A) =>
-    model.decode(data);
+  <A, B>(model: Decoder<A, B>, notification?: () => void) =>
```

Comment:
This is not a good API.

1. callback should becalled something else than "notification". Maybe `onError`?
2. Callback should maybe give error on the callback to the caller:
`onError: (error: E) => void`

Do u really need callback hook here? This could probably be handled on the callee side?

Reply:
Also naming ties this to the place this is used
--> Should be defined there then :).